src/bindings/js/node/src/infer_request.cpp

```diff
@@ -190,3 +197,82 @@ void InferRequestWrap::infer(const Napi::Object& inputs) {
 Napi::Value InferRequestWrap::get_compiled_model(const Napi::CallbackInfo& info) {
     return CompiledModelWrap::Wrap(info.Env(), _infer_request.get_compiled_model());
 }
+
+struct TsfnContext {
+    TsfnContext(Napi::Env env) : deferred(Napi::Promise::Deferred::New(env)){};
+
+    Napi::Promise::Deferred deferred;
+    ov::InferRequest* _context_ir;
+    std::thread nativeThread;
+
+    Napi::ThreadSafeFunction tsfn;
+    std::vector<ov::Tensor> input_tensors;
+    std::map<std::string, ov::Tensor> result;
+};
+
+void FinalizerCallback(Napi::Env env, void* finalizeData, TsfnContext* context) {
+    context->nativeThread.join();
+    delete context;
+};
+
+void performInferenceThread(TsfnContext* data) {
+    infer_mutex.lock();
+    for (size_t i = 0; i < data->input_tensors.size(); ++i) {
+        data->_context_ir->set_input_tensor(i, data->input_tensors[i]);
+    }
+    data->_context_ir->infer();
+
+    auto compiled_model = data->_context_ir->get_compiled_model().outputs();
+    std::map<std::string, ov::Tensor> outputs;
```

Comment:
```suggestion
    std::unordered_map<std::string, ov::Tensor> outputs;
```
If order of elements is not important use unordered map is faster.

Reply:
It might be important later if similar structure like `OVDict` is introduced. Overall there is no huge overhead anyway as models return only few outputs at most (most of the time...).

Reply:
Creation of std::map is more costly than unordered_map. But if order matters then it can not be changed.
This is suggestion to reconsider use simpler object.